OpenSim/Common/DelimFileAdapter.h

```diff
@@ -60,17 +115,650 @@ class OSIMCOMMON_API DelimFileAdapter : public FileAdapter {
+                                     value);
+              } else if(trimmed_key == _versionString) {
+                // Discard version number. Version number is added during
+                // writing. 
+              } else {
+                table->updTableMetaData().setValueForKey(key, value);
+              }
+              continue;
+            }
+        }
+
+        if(header.empty())
+            header = line;
+        else
+            header += "\n" + line;
+    }
+    table->updTableMetaData().setValueForKey("header", header);
+
+    // Callable to get the next line in form of vector of tokens.
+    auto nextLine = [&] {
+        return getNextLine(in_stream, _delimitersRead);
+    };
+
+    // Read the line containing column labels and fill up the column labels
+    // container.
+    auto column_labels = nextLine();
+    ++line_num;
+    // Column 0 is the time column. Check and get rid of it. The data in this
+    // column is maintained separately from rest of the data.
+    OPENSIM_THROW_IF(column_labels[0] != _timeColumnLabel,
+                     UnexpectedColumnLabel,
+                     fileName,
+                     _timeColumnLabel,
+                     column_labels[0]);
+    column_labels.erase(column_labels.begin());
+    // Set the column labels as metadata.
+    ValueArray<std::string> value_array{};
+    for(const auto& cl : column_labels)
+        value_array.upd().push_back(SimTK::Value<std::string>{cl});
+    typename TimeSeriesTable_<T>::DependentsMetaData dep_metadata{};
+    dep_metadata.setValueArrayForKey("labels", value_array);
+    table->setDependentsMetaData(dep_metadata);
+
+    // Read the rows one at a time and fill up the time column container and
+    // the data container.
+    auto row = nextLine();
+    while(!row.empty()) {
+        ++line_num;
+
+        // Time is column 0.
+        double time = std::stod(row.front());
+        row.erase(row.begin());
+
+        auto row_vector = readElems(row);
+
+        OPENSIM_THROW_IF(row_vector.size() != column_labels.size(),
+                         RowLengthMismatch,
+                         fileName,
+                         line_num,
+                         column_labels.size(),
+                         static_cast<size_t>(row_vector.size()));
+
+        // Column 1 is time.
+        table->appendRow(time, std::move(row_vector));
+
+        row = nextLine();
+    }
+
+    OutputTables output_tables{};
+    output_tables.emplace(_table, table);
+
+    return output_tables;
+}
+
+template<>
+inline SimTK::RowVector_<double>
+DelimFileAdapter<double>::readElems(
+                                 const std::vector<std::string>& tokens) const {
+    SimTK::RowVector_<double> elems{static_cast<int>(tokens.size())};
+    for(auto i = 0u; i < tokens.size(); ++i)
+        elems[static_cast<int>(i)] = std::stod(tokens[i]);
+
+    return elems;
+}
+
+template<>
+inline SimTK::RowVector_<SimTK::Vec2>
+DelimFileAdapter<SimTK::Vec2>::readElems(
```

Comment:
If you were clear on the goals of this PR from the beginning, it would have saved me a lot of my time writing and re-writing code. I submit the PR and you decide you want to support more types. I wrote the implementation for the initial goal.

I will change the implementation to partial specialization and get back. Before, I do that I need to know EXACTLY what the goals of this PR are. I am not interested in keep re-doing things.